### PR TITLE
Comment fish

### LINKED ISSUE #152: unit test warnings
It'd be good to get rid of the warning from the unit tests.
```
jeff@dean ~/git/scVI $ make test
python3 setup.py test
running pytest
running egg_info
creating scvi.egg-info
writing scvi.egg-info/PKG-INFO
writing dependency_links to scvi.egg-info/dependency_links.txt
writing requirements to scvi.egg-info/requires.txt
writing top-level names to scvi.egg-info/top_level.txt
writing manifest file 'scvi.egg-info/SOURCES.txt'
reading manifest file 'scvi.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
writing manifest file 'scvi.egg-info/SOURCES.txt'
running build_ext
===================================================================== test session starts =====================================================================
platform linux -- Python 3.6.6, pytest-3.4.2, py-1.5.3, pluggy-0.6.0
rootdir: /home/jeff/git/scVI, inifile: setup.cfg
plugins: nbval-0.9.1
collected 24 items                                                                                                                                            

tests/test_scvi.py ........................                                                                                                             [100%]

====================================================================== warnings summary =======================================================================
tests/test_scvi.py::test_synthetic_1
  /home/jeff/miniconda3/lib/python3.6/site-packages/sklearn/metrics/pairwise.py:257: RuntimeWarning: invalid value encountered in sqrt
    return distances if squared else np.sqrt(distances, out=distances)

tests/test_scvi.py::test_pbmc
  /home/jeff/git/scVI/scvi/dataset/dataset.py:170: RuntimeWarning: divide by zero encountered in log
    log_counts = np.log(X.sum(axis=1))
  /home/jeff/miniconda3/lib/python3.6/site-packages/numpy/core/_methods.py:112: RuntimeWarning: invalid value encountered in subtract
    x = asanyarray(arr - arrmean)

-- Docs: http://doc.pytest.org/en/latest/warnings.html
=========================================================== 24 passed, 3 warnings in 83.57 seconds ============================================================
```

### LINKED ISSUE #171: docstrings for vae_fish.py
We're missing docstrings current for the methods in `vae_fish.py`.

- [ ] get_latents
- [ ] sample_from_posterior_z
- [ ] sample_from_posterior_l
- [ ] get_sample_scale
- [ ] get_sample_rate
- [ ] get_sample_rate_fish
- [ ] classify
- [ ] forward

We'd like to have docstrings that look more like this one:
https://github.com/YosefLab/scVI/blob/8147cb6b11717adb7578cf95c97c8ef5ae3c3386/scvi/models/modules.py#L42-L50

### PR DESCRIPTION
**Pull request by jeff-regier**
closes #171 
closes #152

### CODE REVIEW

File: scvi/inference/posterior.py [hunk 1]
```diff
@@ -544,7 +544,7 @@ def nn_overlap(X1, X2, k=100):
     Compute the overlap between the k-nearest neighbor graph of X1 and X2 using Spearman correlation of the
     adjacency matrices.
     '''
-    nne = NearestNeighbors(n_neighbors=k + 1, n_jobs=8)
+    nne = NearestNeighbors(n_neighbors=k + 1)# "n_jobs=8
```
Comment (jeff-regier) [inline scvi/inference/posterior.py]: @Edouard360 Is this change OK? I'm not familiar with what `n_jobs` did.

File: requirements.txt [hunk 2]
```diff
@@ -1,11 +1,11 @@
-pip==10.0.1
+pip>=10.0.1
```
Comment (jeff-regier) [inline requirements.txt]: Better to specify a particular version number here that is known to work. (These versions get used by travis-ci.) 

It's in setup.py where we specify a range of versions that our software should work with.

File: scvi/dataset/dataset.py [hunk 3]
```diff
@@ -120,11 +120,12 @@ def update_genes(self, subset_genes):
         if hasattr(self, 'gene_symbols'):
             self.gene_symbols = self.gene_symbols[subset_genes]
         self.nb_genes = self.X.shape[1]
+        self.update_cells(np.array(self.X.sum(axis=1)>0).ravel())
 
     def update_cells(self, subset_cells):
         new_n_cells = len(subset_cells) if subset_cells.dtype is not np.dtype('bool') else subset_cells.sum()
         print("Downsampling from %i to %i cells" % (len(self), new_n_cells))
-        for attr_name in ['_X', 'local_means', 'local_vars', 'labels', 'batch_indices']:
+        for attr_name in ['_X', 'labels', 'batch_indices', 'local_means', 'local_vars']:
```
Comment (jeff-regier) [inline scvi/dataset/dataset.py]: Does the reordering matter?

Comment (jules-samaran) [reply scvi/dataset/dataset.py]: No it doesn't, it's because I had suppressed some and didn't think about the order when I rewrited them in the code

**Comment by jeff-regier** [comment]
Looks like a travis error.

File: requirements.txt (hunk 2 above)
Comment (jules-samaran) [reply requirements.txt]: I did that because on my computer pip 10.0.1 would not allow me to do things a more recent version (18.0 I think) did

**Pull request merged**